Content.Client/VoiceMask/VoiceMaskNameChangeWindow.xaml.cs

```diff
@@ -28,54 +46,104 @@ public VoiceMaskNameChangeWindow()
+        _speechVerbs.Sort((a, b) => a.Item1.CompareTo(b.Item1));
+
+        foreach (var noise in proto.EnumeratePrototypes<SpeechSoundsPrototype>())
+        {
+            _speechSounds.Add((Loc.GetString(noise.Name), noise.ID));
+        }
+        _speechSounds.Sort((a, b) => a.Item1.CompareTo(b.Item1));
     }
 
-    public void AddVerbs()
+    /// <summary>
+    ///     Populates all the buttons with the values stored in the lists.
+    /// </summary>
+    public void AddVerbsAndSounds()
     {
-        SpeechVerbSelector.Clear();
+        PopulateOptionButton(SpeechVerbSelector, _selectedVerb, _speechVerbs);
+        PopulateOptionButton(SpeechSoundSelector, _selectedSound, _speechSounds);
+    }
 
-        AddVerb(Loc.GetString("chat-speech-verb-name-none"), null);
-        foreach (var (name, id) in _verbs)
-        {
-            AddVerb(name, id);
-        }
+    /// <summary>
+    ///     Updates the window to the current state. E.g make sure the shown name is the actual name being used etc..
+    /// </summary>
+    public void UpdateState(VoiceMaskBuiState voiceMaskBuiState)
+    {
+        NameSelector.Text = voiceMaskBuiState.Name;
+
+        _selectedVerb = voiceMaskBuiState.Verb;
+        _selectedSound = voiceMaskBuiState.Sound;
+
+        UpdateSelectedButtonOption(SpeechVerbSelector, _selectedVerb);
+        UpdateSelectedButtonOption(SpeechSoundSelector, _selectedSound);
     }
 
-    private void AddVerb(string name, string? verb)
+    #region Helper functions
+
+    /// <summary>
+    ///     Populates a button with the given list of names / ids. Also adds a "default" button with null as the id.
+    /// </summary>
+    private void PopulateOptionButton(OptionButton button, string? selectedOption, List<(string, string)> values)
     {
-        var id = SpeechVerbSelector.ItemCount;
-        SpeechVerbSelector.AddItem(name);
-        if (verb is {} metadata)
-            SpeechVerbSelector.SetItemMetadata(id, metadata);
+        button.Clear();
 
-        if (verb == _verb)
-            SpeechVerbSelector.SelectId(id);
+        // Add the default option that wont do anything when selected.
+        AddOption(button, selectedOption, Loc.GetString("chat-speech-verb-name-none"), null);
+        foreach (var (name, id) in values)
+        {
+            AddOption(button, selectedOption, name, id);
+        }
     }
 
-    public void UpdateState(string name, string? verb)
+    /// <summary>
+    ///     Actually adds the option to the drop down menu. If the given value matches the selected value, it will also
+    ///     make the drop down menu display it as the currently selected value.
+    /// </summary>
+    private void AddOption(OptionButton button, string? selectedValue, string name, string? value)
     {
-        NameSelector.Text = name;
-        _verb = verb;
+        var id = button.ItemCount;
+        button.AddItem(name);
+        if (value is { } metadata)
+            button.SetItemMetadata(id, metadata);
```

Comment:
Equivalent and less confusing: 

```suggestion

        if (value is not null)
            button.SetItemMetadata(id, value);
```